rbd/rbd_test.go

```diff
@@ -578,6 +585,31 @@ func TestWriteSame(t *testing.T) {
 		assert.Equal(t, int64(0), n_in)
 	})
 
+	t.Run("zerofill", func(t *testing.T) {
+		// implement allocating an image by writing zeros to it
+		sc, err := img.GetStripeCount()
+		assert.NoError(t, err)
+
+		// fetch the actual size of the image as available in the pool, can be
+		// margially different from the requested image size
+		st, err := img.Stat()
+		assert.NoError(t, err)
+		size := st.Size
+
+		// zeroBlock is the stripe-period: size of the object-size multiplied
+		// by the stripe-count
+		order := uint(st.Order) // signed shifting requires newer golang?!
+		zeroBlock := make([]byte, sc*(1<<order))
+		n_in, err := img.WriteSame(0, size, zeroBlock, rados.OpFlagNone)
+		assert.NoError(t, err)
+		assert.Equal(t, size, uint64(n_in))
+
+		cmd := exec.Command("rbd", "diff", "--pool", poolname, name)
```

Comment:
It seems a bit weird to exec the rbd tool here when we are an API for this stuff. :-)
Isn't rbd diff based on diff_iterate? (I just checked, yes, diff_iterate2, which we already wrap!)
Would it be much of a burden if I asked to reimplement this part using go-ceph's DiffIterate?

Reply:
I think I remember @nixpanic explicitly mentioned something like using the tool to have different code verifying the behavior?

Reply:
Yes, I know we have DiffIterate, but it would add quite some additional code to the test, which is not _that_ relevant. If consuming DiffIterate is much preferred, I could look into that (at a later time).

Reply:
So my main issue is that it adds an external dependency to the test. It may not be common to run the tests outside the CI container, in an environment where rbd command line is not installed but it could happen. I'm fine with deferring it until later but I would like to see it. Would you file an issue to track that or would you like me to?

Reply:
You can also see it that way: these are "just" the tests, whose main purpose are to be run in our CI. Also it's quite unlikely that people have set up their own testing cluster, but don't have rbd installed, and can't install it with a simple install command. Do we really want to invest our time for cases, that are very unlikely to exist? There is nothing stopping people, who actually face that issue, from filing a PR, if they are inclined to. So I see this as a non-issue in the real-world.